src/main/java/com/zufar/icedlatte/order/api/OrderProvider.java

```diff
@@ -0,0 +1,89 @@
+package com.zufar.icedlatte.order.api;
+
+import com.zufar.icedlatte.openapi.dto.*;
+import com.zufar.icedlatte.order.converter.OrderDtoConverter;
+import com.zufar.icedlatte.order.entity.Order;
+import com.zufar.icedlatte.order.entity.OrderItem;
+import com.zufar.icedlatte.order.repository.OrderRepository;
+import com.zufar.icedlatte.product.repository.ProductInfoRepository;
+import com.zufar.icedlatte.security.api.SecurityPrincipalProvider;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.OffsetDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.zufar.icedlatte.order.api.Calculator.calculate;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class OrderProvider {
+
+    private final OrderRepository orderRepository;
+    private final ProductInfoRepository productInfoRepository;
+    private final OrderDtoConverter orderDtoConverter;
+    private final SecurityPrincipalProvider securityPrincipalProvider;
+    public static final int DEFAULT_PRODUCTS_QUANTITY = 0;
+
+    @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.READ_COMMITTED)
+    public AddedOrder addNewOrder(final OrderDto orderBody) {
```

Comment:
Class has suffix Provider. It means it should not have methods like `addNewOrder`. Put `addNewOrder` to another service class like OrderAdder.

Reply:
Done in [46e65b4](https://github.com/Sunagatov/Iced-Latte/pull/217/commits/46e65b4c9646349a131383ace42afdeddba4b663)